### DISCUSSION TITLE
netty channel exception using pushy in tomcat application

### DISCUSSION THREAD

**Discussion by rswiston** [relayrides/pushy#1026] [Q&A]
Greetings,

I'm trying to use pushy in a tomcat application.  I've written a stand-alone test of my pushy code, and it works fine.  But when the same code executes in my tomcat application, ApnsChannelPool reports "Failed to create channel" with this detail:

io.netty.channel.StacklessClosedChannelException: null
        at io.netty.channel.AbstractChannel$AbstractUnsafe.ensureOpen(ChannelPromise)(Unknown Source) ~[netty-transport-4.1.86.Final.jar:4.1.86.Final]

I'm using pushy 0.15.2.  I'm happy to provide any additional details that would help anyone to help me. I've enabled trace logging, but I'm not sure what to do next to diagnose.

Suggestions appreciated - many thanks!

**Comment by rswiston** [comment]
Sorry, please ignore the above - it was user error.  I failed to include a netty library in the WAR file used by tomcat.

**Reply by jchambers** [reply]
Glad you got it sorted out, and thank you for following up so other folks can learn from what you found!
